service/history/engine/interface.go

```diff
@@ -80,8 +80,8 @@ type (
 		DescribeCrossClusterQueue(ctx context.Context, clusterName string) (*types.DescribeQueueResponse, error)
 
 		NotifyNewHistoryEvent(event *events.Notification)
-		NotifyNewTransferTasks(executionInfo *persistence.WorkflowExecutionInfo, tasks []persistence.Task)
-		NotifyNewTimerTasks(executionInfo *persistence.WorkflowExecutionInfo, tasks []persistence.Task)
-		NotifyNewCrossClusterTasks(executionInfo *persistence.WorkflowExecutionInfo, tasks []persistence.Task)
+		NotifyNewTransferTasks(executionInfo *persistence.WorkflowExecutionInfo, tasks []persistence.Task, persistenceError bool)
```

Comment:
Should we add an entity that contains this flag instead? This will help extend in the future without modifying the interface.

Reply:
That brings a question of where to put the definition of the struct. It's not a simple change because it might introduce circular dependencies.